1단계 - 엔티티 매핑

onederacy opened the pull request:
안녕하세요. 리뷰어님 

그런데 JPA를 눈으로만 보다가 직접 사용해본건 처음이네요...
JPA 관련 테스트를 만들때는 기존 repository에 있는 메소드만 검증하면 되는건지,
아니면 CRUD에 관해서는 다 만들어야 하는건가요? 

아직 1단계라 큰 어려움은 없었습니다.
enum -> DB column 관련 내용은 이전에 우형 블로그에서 본게 있어서 다행이었네요 ㅎㅎ

부족한 부분 리뷰 팍팍 부탁드립니다 (_ _)

Commit:
refactor:
- User entity mapping
- add User test with DataJpaTest

Commit:
refactor:
 - Question entity mapping
 - add Question test with DataJpaTest

Commit:
refactor: refactor QuestionTest class

Commit:
refactor:
- Answer entity mapping
- add AnswerTest with DataJpaTest

Commit:
refactor:
 - DeleteHistory entity mapping
 - add DeleteHistoryTest with DataJpaTest
 - add ContentTypeConverter for entity mapping

kingbbode commented on src/main/java/qna/domain/converter/ContentTypeConverter.java:
```diff
@@ -0,0 +1,21 @@
+import javax.persistence.Converter;
+
+@Converter
+public class ContentTypeConverter implements AttributeConverter<ContentType,String> {
```
`type` 값이 필요한 이유는 무엇인가요?

`@Enumerated` 를 찾아보시면 도움이 될 듯 합니다 :)

onederacy replied:
기존 ```converter``` 예제를 적용하려다보니 불필요하게 type이라는 변수를 사용하게 되었네요...
만약 converter를 사용할 시, ```enum```의 ```name()``` 메소드로 리턴하면 되는 부분 확인 하였습니다.

```@Enumrated```에 대해 몰랐는데 단순히 enum의 값을 적용할 때는 converter보다 편할 것 같습니다.

감사합니다.(_ _)

kingbbode reviewed (changes requested):
코멘트 작성해두었습니다 :)
확인 부탁드려요.

Commit:
refactor: refactoring about reviewer's comments
- refactor DeleteHistory class about @Converter -> @Enumerated
- delete unnecessary param in ContentType
- use ContentType.name() in ContentTypeConverter

kingbbode reviewed (approved):
다음 단계도 잘 부탁드립니다 :)
